src/translations/indexMapper.js

```diff
@@ -336,52 +527,114 @@ class IndexMapper {
    *
    * @private
    * @param {boolean} [readFromCache=true] Determine if read indexes from cache.
    * @returns {Array}
    */
-  getFlattenSkipList(readFromCache = true) {
+  getFlattenTrimmingResult(readFromCache = true) {
     if (readFromCache === true) {
-      return this.flattenSkipList;
+      return this.flattenTrimmingResult;
     }
 
-    if (this.skipMapsCollection.getLength() === 0) {
+    if (this.trimmingMapsCollection.getLength() === 0) {
       return [];
     }
 
     const result = [];
-    const particularSkipsLists = arrayMap(this.skipMapsCollection.get(), skipList => skipList.getValues());
-
-    rangeEach(this.indexesSequence.getLength(), (physicalIndex) => {
-      result[physicalIndex] = particularSkipsLists.some(particularSkipsList => particularSkipsList[physicalIndex]);
+    // Below variable stores results of trimming (boolean values) for every particular trimming map.
+    // [
+    //   [true, true, true], // first trimming map
+    //   [false, true, true] // second trimming map
+    // ]
+    const allTrimmingValues = arrayMap(this.trimmingMapsCollection.get(), trimmingMap => trimmingMap.getValues());
+
+    rangeEach(this.indexesSequence.getLength() - 1, (physicalIndex) => {
+      result[physicalIndex] = allTrimmingValues.some(nextTrimmingValues => nextTrimmingValues[physicalIndex]);
     });
 
     return result;
   }
 
   /**
-   * Rebuild cache for some indexes. Every action on indexes sequence or skipped indexes by default reset cache,
-   * thus batching some index maps actions is recommended.
+   * Get flat list of values, which are result whether index has been hidden in any registered hiding maps.
    *
-   * @param {boolean} [force=false] Determine if force cache update.
    * @private
+   * @param {boolean} [readFromCache=true] Determine if read indexes from cache.
+   * @returns {Array}
    */
-  updateCache(force = false) {
-    if (force === true || (this.isBatched === false && this.cachedIndexesChange === true)) {
-      this.flattenSkipList = this.getFlattenSkipList(false);
-      this.notSkippedIndexesCache = this.getNotSkippedIndexes(false);
-      this.cachedIndexesChange = false;
+  getFlattenHidingResult(readFromCache = true) {
+    if (readFromCache === true) {
+      return this.flattenHidingResult;
+    }
 
-      this.runLocalHooks('cacheUpdated');
+    if (this.hidingMapsCollection.getLength() === 0) {
+      return [];
     }
+
+    const result = [];
+    // Below variable stores results of hiding (boolean values) for every particular hiding map.
+    // [
+    //   [true, true, true], // first trimming map
+    //   [false, true, true] // second trimming map
+    // ]
+    const allHidingValues = arrayMap(this.hidingMapsCollection.get(), list => list.getValues());
+
+    rangeEach(this.indexesSequence.getLength() - 1, (physicalIndex) => {
+      result[physicalIndex] = allHidingValues.some(nextHidingValues => nextHidingValues[physicalIndex]);
+    });
+
+    return result;
```

Comment:
This observation is correct - the code is duplicated and I missed that. For me, the `MapCollection` shouldn't have an abstract method as it is a fully useful being, not the abstract one. I have thought about the name `BooleanMapCollection` and for me, this describes the character of our collection but not the idea of what we do. Maybe another name will be a proper answer to our problem? Btw. Maybe there should be used a mixin rather than an inheritance? WDYT?

My proposition: https://github.com/handsontable/handsontable/pull/6547/commits/a5a2593707e5be54160c8d216b6b4abb46342457#diff-3c59964e4bb5213ed3243b4b54ab0172